### PR TITLE
Bte merchant dash orders us 35 issue #39

### LINKED ISSUE #39: User Story 35, Merchant Dashboard displays Orders
```
As a merchant
When I visit my dashboard ("/dashboard")
If any users have pending orders containing items I sell
Then I see a list of these orders.
Each order listed includes the following information:
- the ID of the order, which is a link to the order show page ("/dashboard/orders/15")
- the date the order was made
- the total quantity of my items in the order
- the total value of my items for that order
```

DO THESE STORIES FIRST:
- User Story 34, Merchant Dashboard Show Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #41

child of #82

depends on #40

### PR DESCRIPTION
**Pull request by blake-enyart**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 97.86%%

# What functionality does this accomplish?
Create pending orders list on merchant dashboard

# What did you struggle on to complete?
* Test development on both the features and models. I would like some review here.
* FactoryBot associations

# Implements/Fixes:
* description
closes #39 

# Testing Changes
- [X] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [X] My code has no unused/commented out code
- [X] I have reviewed my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code

# Please include an emoji of how you feel about this branch:
😠

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -6,4 +6,13 @@ class Order < ApplicationRecord
   has_many :items, through: :order_items
 
   enum status: ['pending', 'packaged', 'shipped', 'cancelled']
+
+  def total_quantity
+    self.items.sum(:quantity)
```
Comment (milevy1) [inline app/models/order.rb]: I think we want to sum the `order_items.quantity`

The way you have it sums the `items.quantity` which is the _stock available from the merchant._

Comment (milevy1) [reply app/models/order.rb]: Why did we name both of them quantity 🤷‍♂️ haha

File: spec/models/order_spec.rb [hunk 2]
```diff
@@ -10,4 +10,36 @@
...
+      oi_1 = create(:order_item)
+      #creates three order_items and items bound to same order from oi_1
+      oi_2, oi_3, oi_4 = create_list(:order_item, 3, order: oi_1.order)
+      #main order for clarity in test
+      order_1 = oi_1.order
+
+      #total quantity of items in a single order
+      actual = order_1.total_quantity
+      #generated via pry session AR query
+      expected = 28
```
Comment (milevy1) [inline spec/models/order_spec.rb]: We should not hard code this.  It's probably why the test is passing even though .total_quantity is calling stock of items

File: spec/models/order_spec.rb [hunk 3]
```diff
@@ -10,4 +10,36 @@
...
+      oi_1 = create(:order_item)
+      #creates three order_items and items bound to same order from oi_1
+      oi_2, oi_3, oi_4 = create_list(:order_item, 3, order: oi_1.order)
+      #main order for clarity in test
+      order_1 = oi_1.order
+
+      #grand total of item prices summed up for single order
+      actual = order_1.grand_total
+      #generated via pry session AR query
+      expected = 162.0
```
Comment (milevy1) [inline spec/models/order_spec.rb]: Same thing here, we should not hard code our test expectations

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -34,27 +34,41 @@
...
+      expect(item_2.enabled).to eq(false)
+      expect(item_3.enabled).to eq(false)
+    end
+
+    it '#pending_orders' do
+      merchant = create(:merchant)
+      #creates 3 order_items and 3 associated items/pending orders
+      oi_1, oi_2, oi_3 = create_list(:order_item, 3)
+      #creates second item associated with same order
+      oi_4 = create(:order_item, order_id: oi_1.order.id, order: oi_1.order)
```
Comment (milevy1) [inline spec/models/user_spec.rb]: Why does an :order_item need both an order_id and an order?
